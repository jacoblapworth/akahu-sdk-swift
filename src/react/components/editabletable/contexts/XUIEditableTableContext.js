import React from 'react';

export default React.createContext({
  rowOptions: {
    isRemovable: false,
  },
});
