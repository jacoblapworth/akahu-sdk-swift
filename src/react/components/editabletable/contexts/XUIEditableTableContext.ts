import * as React from 'react';

interface XUIEditableTableContext {
  columnWidths?: string[];
  dragAndDrop: {
    dragHandleDescribedBy?: string;
    rowPlaceholder?: React.ReactElement<HTMLElement> | null;
  };
  rowOptions: {
    dragButtonAriaLabel?: string;
    isDraggable?: boolean;
    isRemovable?: boolean;
    removeButtonAriaLabel?: string;
  };
  scrollContainerRef?: React.MutableRefObject<HTMLElement>;
  tableRef?: React.MutableRefObject<HTMLTableElement>;
  tableWrapperRef?: React.MutableRefObject<HTMLElement>;
}

export default React.createContext<XUIEditableTableContext>({
  dragAndDrop: {},
  rowOptions: {},
});
