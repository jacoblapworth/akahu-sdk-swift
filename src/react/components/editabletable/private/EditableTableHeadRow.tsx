import * as PropTypes from 'prop-types';
import * as React from 'react';

import XUIEditableTableContext from '../contexts/XUIEditableTableContext';
import XUIEditableTableHeadingCell from '../XUIEditableTableHeadingCell';
import { xuiControlSizeStandard } from './constants';

interface BaseProps {
  qaHook?: string;
}

type Props = BaseProps & React.HTMLAttributes<HTMLTableRowElement>;

const EditableTableHeadRow: React.FunctionComponent<Props> = ({
  children,
  qaHook,
  ...spreadProps
}) => {
  const {
    rowOptions: { isDraggable, isRemovable },
  } = React.useContext(XUIEditableTableContext);

  return (
    <tr data-automationid={qaHook} {...spreadProps}>
      {isDraggable && <XUIEditableTableHeadingCell style={{ width: xuiControlSizeStandard }} />}
      {children}
      {isRemovable && <XUIEditableTableHeadingCell style={{ width: xuiControlSizeStandard }} />}
    </tr>
  );
};

EditableTableHeadRow.propTypes = {
  children: PropTypes.node,
  qaHook: PropTypes.string,
};

export default EditableTableHeadRow;
