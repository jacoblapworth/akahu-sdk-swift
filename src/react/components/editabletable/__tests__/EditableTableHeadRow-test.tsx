import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { axe, toHaveNoViolations } from 'jest-axe';
import React from 'react';

import XUIEditableTableContext from '../contexts/XUIEditableTableContext';
import EditableTableHeadRow from '../private/EditableTableHeadRow';

Enzyme.configure({ adapter: new Adapter() });
expect.extend(toHaveNoViolations);

describe('EditableTableHeadRow', () => {
  it('renders correctly', () => {
    // Arrange
    const wrapper = shallow(<EditableTableHeadRow />);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with children, qaHook, and spreadprops', () => {
    // Arrange
    const wrapper = shallow(
      <EditableTableHeadRow className="test-classname" qaHook="test-qaHook">
        <td>Test content</td>
      </EditableTableHeadRow>,
    );

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders utility control columns, if applicable', () => {
    // Arrange
    const wrapper = mount(
      <XUIEditableTableContext.Provider
        value={{ dragAndDrop: {}, rowOptions: { isDraggable: true, isRemovable: true } }}
      >
        <table>
          <thead>
            <EditableTableHeadRow />
          </thead>
        </table>
      </XUIEditableTableContext.Provider>,
    );

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should pass accessibility testing', async () => {
    const wrapper = mount(
      <table>
        <thead>
          <EditableTableHeadRow />
        </thead>
      </table>,
    );
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });
});
