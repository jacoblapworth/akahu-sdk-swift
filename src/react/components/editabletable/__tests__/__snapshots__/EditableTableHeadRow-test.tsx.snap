// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTableHeadRow renders correctly 1`] = `<tr />`;

exports[`EditableTableHeadRow renders utility control columns, if applicable 1`] = `
<table>
  <thead>
    <EditableTableHeadRow>
      <tr>
        <XUIEditableTableHeadingCell
          inlineAlignment="start"
          scope="col"
          style={
            Object {
              "width": "40px",
            }
          }
        >
          <th
            className="xui-editabletableheadingcell xui-editabletableheadingcell-leftaligned"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            scope="col"
            style={
              Object {
                "width": "40px",
              }
            }
          />
        </XUIEditableTableHeadingCell>
        <XUIEditableTableHeadingCell
          inlineAlignment="start"
          scope="col"
          style={
            Object {
              "width": "40px",
            }
          }
        >
          <th
            className="xui-editabletableheadingcell xui-editabletableheadingcell-leftaligned"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            scope="col"
            style={
              Object {
                "width": "40px",
              }
            }
          />
        </XUIEditableTableHeadingCell>
      </tr>
    </EditableTableHeadRow>
  </thead>
</table>
`;

exports[`EditableTableHeadRow renders with children, qaHook, and spreadprops 1`] = `
<tr
  className="test-classname"
  data-automationid="test-qaHook"
>
  <td>
    Test content
  </td>
</tr>
`;
